include/hw_constants_checks.h

```diff
@@ -16,12 +16,20 @@ namespace gbt{
      *  \param[in] phase Phase value to check.
      *  \return Returns `false` in case of success; `true` in case of error. The precise error is logged and written to the `error` RPC key.
      */
-    inline bool checkPhase(uint8_t phase){
+    inline void checkPhase(uint8_t phase){
         if (phase < PHASE_MIN)
-            EMIT_RPC_ERROR(stdsprintf("The phase parameter supplied (%hhu) is smaller than the minimal phase (%hhu).", phase, PHASE_MIN), true)
+            std::stringstream errmsg;
+            errmsg.clear();
```

Comment:
for a new `stringstream`, you don't need the `clear()` and `str("")` calls, this is only needed if you are reusing the same `stringstream` object later within the scope